### ISSUE TITLE
x86 16bit on ARM64

### ISSUE THREAD

**Issue by Rorda** [otya128/winevdm#1280]
I believe otvdm is an excellent piece of work.

Is there any chance you could write something similar for the ARM64 processor?
I know, running a x86 16bit piece of software on a ARM64 sounds crazy.

never hurts to ask....

**Comment by darkhog** [comment]
It would require full CPU emulation, which is beyond the scope of this project and would have a terrible performance anyway.

**Comment by DarkShadow44** [comment]
> It would require full CPU emulation, which is beyond the scope of this project and would have a terrible performance anyway.

AFAIK it already uses MAME for 16bit CPU emulation. You can't run 16bit code on 64bit windows.

**Comment by kristibektashi** [comment]
This is a duplicate of #1038

**Comment by kristibektashi** [comment]
> It would require full CPU emulation, which is beyond the scope of this project and would have a terrible performance anyway.

It already uses CPU emulation even on x86, since the x64 version of Windows that it's meant for disallows access to the necessary 16-bit protected mode. Also, the x64 version works fine within ARM64 Windows, but you're emulating the emulator itself (which is strictly x86-32), so you're running in double emulation. But as a proof of concept, it works well enough

**Comment by darkhog** [comment]
> > It would require full CPU emulation, which is beyond the scope of this project and would have a terrible performance anyway.
> 
> It already uses CPU emulation even on x86, since the x64 version of Windows that it's meant for disallows access to the necessary 16-bit protected mode. Also, the x64 version works fine within ARM64 Windows, but you're emulating the emulator itself (which is strictly x86-32), so you're running in double emulation. But as a proof of concept, it works well enough

Yeah, that's what I meant. Emulatorception. Not very performant and you really should just emulate 16-bit x86 directly in ARM if you want to do this.

**Comment by kristibektashi** [comment]
> > > It would require full CPU emulation, which is beyond the scope of this project and would have a terrible performance anyway.
> > 
> > 
> > It already uses CPU emulation even on x86, since the x64 version of Windows that it's meant for disallows access to the necessary 16-bit protected mode. Also, the x64 version works fine within ARM64 Windows, but you're emulating the emulator itself (which is strictly x86-32), so you're running in double emulation. But as a proof of concept, it works well enough
> 
> Yeah, that's what I meant. Emulatorception. Not very performant and you really should just emulate 16-bit x86 directly in ARM if you want to do this.

Yes I agree but that has other issues, OTVDM itself doesn't compile for anything other than x86-32, I tried compiling for ARM32 once (for an unrelated port to the Surface RT) and it showed many compiler errors. The same is most definitely also true for ARM64 and maybe even x64, since I believe OTVDM uses a lot of x86 assembly in it's code.
